mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerPerformer.java

```diff
@@ -70,16 +68,17 @@ public ReIndexerPerformer(MailboxManager mailboxManager,
 
     Task.Result reIndex(MailboxPath path, ReprocessingContext reprocessingContext) throws MailboxException {
         MailboxSession mailboxSession = mailboxManager.createSystemSession(path.getUser());
-        return reIndex(path, mailboxSession, reprocessingContext);
+        Mailbox mailbox = mailboxSessionMapperFactory.getMailboxMapper(mailboxSession).findMailboxByPath(path);
+        return reIndexSingleMailbox(mailbox.getMailboxId(), reprocessingContext);
     }
 
     Task.Result reIndex(ReprocessingContext reprocessingContext) throws MailboxException {
-        MailboxSession mailboxSession = mailboxManager.createSystemSession("re-indexing");
+        MailboxSession mailboxSession = mailboxManager.createSystemSession(RE_INDEXING);
         LOGGER.info("Starting a full reindex");
-        List<MailboxPath> mailboxPaths = mailboxManager.list(mailboxSession);
+        List<Mailbox> mailboxes = mailboxSessionMapperFactory.getMailboxMapper(mailboxSession).list();
```

Comment:
This looks suspicious to me that we require the mapper layer. A quick look at Jmap make me think that we should rather use `mailboxManager.search` here. WDYT ?

Reply:
In term of design, we should refrain from using mapper almost everywhere, we will probably have to remove the mapper layer in the future.

Reply:
To answer you, it seems really hard to set up this feature on top of `MailboxManager` for very good reasons:
 - API level: Indexers expects MailboxMessage, which is not part of mailbox API, but only defined at the mapper level. Thus we are forced at some point to use mappers.
 - Rights: We do not get yet "administrative" mailbox session (only impersonation). The bad thing with IDs is that you don't know the user, thus you don't know which username to be impersonating to. This is the issue removed in https://github.com/linagora/james-project/pull/1948/commits/1f359c96fabf164293ca04132ea5a88ab79bf39f

Also, the ReIndexer is only reading from the main store so calling it is safe - we won't encoutered the bugs faced 2.5 years ago with JMAP that an intern did spotted.

Finally, the overall idea of reIndexing is internal to the mailbox (indexing APIs are not exposed outside of this context) hence it does not shock me.

Anyway, acting on any of the two points I listed above is **clearly out of reach** of a 2 point ticket.

Anyway, I am going to split the PR in 2, but I still believe in what I explained here.

Reply:
Ok, got it: for this specific case we can still use `mailboxManager` as it was done before.